### ISSUE TITLE
Feature Request : Service account and Application Key

### ISSUE THREAD

**Issue by rlaveycal** [DataDog/terraform-provider-datadog#1577]
### Affected Resource(s)

-   datadog_application_key 

### Expected Behavior

`datadog_application_key ` only allows you to create a key for the current user. It would be useful to be able to create one for a service account, and for a `datadog_service_account` resource to exist to allow that to be created as well.

Related data sources should also exist/be supported

**Comment by skarimo** [comment]
Hello, this was closed by: https://github.com/DataDog/terraform-provider-datadog/pull/1490 . It should be available in the next release.

**Comment by giorgiopreviterappro** [comment]
Hi, is this feature actually coming in the next release?

I noticed this PR https://github.com/DataDog/terraform-provider-datadog/pull/1586 reverted the change to `datadog_user` in favour of a new `datadog_service_account`

**Comment by jorhett** [comment]
@skarimo you closed this 7 days after #1490 was reverted by #1586

**Comment by skarimo** [comment]
This is a duplicate of https://github.com/DataDog/terraform-provider-datadog/issues/1266 and is being tracked there
